src/cpu.c

```diff
@@ -501,123 +586,134 @@ static void cpu_commit_num_cpu(gauge_t value) /* {{{ */
   return;
 } /* }}} void cpu_commit_num_cpu */
 
-/* Resets the internal aggregation. This is called by the read callback after
- * each iteration / after each call to cpu_commit(). */
-static void cpu_reset(void) /* {{{ */
-{
-  for (size_t i = 0; i < cpu_states_num; i++)
-    cpu_states[i].has_value = false;
-
-  global_cpu_num = 0;
-} /* }}} void cpu_reset */
-
-/* Legacy behavior: Dispatches the raw derive values without any aggregation. */
-static void cpu_commit_without_aggregation(void) /* {{{ */
-{
+static void commit_cpu_usage(usage_t *u, size_t cpu_num) {
   metric_family_t fam = {
       .name = "system.cpu.time",
+      .help = "Microseconds each logical CPU spent in each state",
+      .unit = "us",
       .type = METRIC_TYPE_COUNTER,
   };
 
   metric_t m = {0};
+  if (cpu_num != CPU_ALL) {
+    char cpu_num_str[64];
+    ssnprintf(cpu_num_str, sizeof(cpu_num_str), "%zu", cpu_num);
+    metric_label_set(&m, label_number, cpu_num_str);
+  }
 
-  for (int state = 0; state < COLLECTD_CPU_STATE_ACTIVE; state++) {
-    metric_label_set(&m, "state", cpu_state_names[state]);
-
-    for (size_t cpu_num = 0; cpu_num < global_cpu_num; cpu_num++) {
-      cpu_state_t *s = get_cpu_state(cpu_num, state);
-
-      if (!s->has_value)
+  if (report_by_state) {
+    for (state_t state = 0; state < STATE_ACTIVE; state++) {
+      derive_t usage = usage_count(u, cpu_num, state);
+      if (usage == -1) {
```

Comment:
I think define for the `-1` error values from the called `usage_*` functions would be good.

Reply:
Done.